OpenAPI/kiota/testing.md

```diff
@@ -0,0 +1,247 @@
+---
+title: Unit testing with Kiota API clients
+description: Learn about writing unit tests with Kiota API clients.
+author: LockTar
+ms.author: vibiret
+ms.topic: conceptual
+date: 07/04/2023
+---
+
+# Unit testing with Kiota API clients
+
+Code were Kiota API clients are used can be tested. In this document is explained how to mock the Microsoft Graph SDK.
+
+## Prerequisite
+
+In this sample the following frameworks/tools are used.
+
+- [Microsoft Graph SDK v5](https://github.com/microsoftgraph/msgraph-sdk-dotnet)
```

Comment:
@jasonjoh I wonder whether we should use one of the get started samples instead?

Reply:
You could also do a link to the [overview](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/overview.md) page?

Reply:
I was referring to this description so it's consistent with the rest of the documentation https://learn.microsoft.com/en-us/openapi/kiota/quickstarts/dotnet#generate-the-api-client

Reply:
Ah ok. Yes that is something you as a team should decide  👍🏻

Reply:
yep, Jason is our technical writer. However it's a holiday in the US today, expect some latency :)

Reply:
Yes, I think it makes sense to incorporate this into the quick start project and consume the code directly from there. I'll take a look at it.

Reply:
@jasonjoh added tests based on this PR in the sample project. See [https://github.com/microsoft/kiota-samples/pull/2854](https://github.com/microsoft/kiota-samples/pull/2854). Will resolve conversation.

Reply:
Thanks for putting the pull request together Jason. Keeping this open as we need to update the snippets here once the pr is merged.

Reply:
Snippet replaced with a snippet reference, it will load the code from the sample project.